rfcs/01924_tls_attestation_ca.md

```diff
@@ -0,0 +1,161 @@
+# RFC#01924: Remote Attestation via TLS (Certificate Authority)
+
+## Objective
+
+Remotely attest applications in Trusted Execution Environments (TEEs) over the
+TLS protocol.
+
+## Background
+
+A lot of existing applications that would benefit from the remote attestation
+use **TLS protocol** to create secure channels with their clients. But
+integrating the remote attestation might bring up unnecessary complications in a
+form of completely rewriting the application connection logic.
+
+Thus, instead of refactoring existing applications and implementing a new
+protocol, we propose to integrate remote attestation into TLS by
+cryptographically binding **TLS certificates** to the **TEE firmware**.
+
+## Overview
+
+In order to allow clients to use pure TLS to connect to TEE applications we
+propose a Proxy Attestation Service.
+
+Proxy Attestation Service - is a service that can be used by clients to remotely
+attest TEE applications. The key idea is: instead of attesting TEE applications
+themselves, clients attest the proxy, which attests TEE applications on behalf
+of the clients.
+
+There are several benefits from implementing the Proxy Attestation Service:
+
+- **The ability to run TLS applications as is**
+  - The proxy will allow clients to use TLS certificates to connect to TEE
+    applications
+  - Since attestation will be handled by the proxy - TEE applications could use
+    legacy TLS code without the need to support remote attestation logic
+- **Support multiple TEEs**
+  - Proxy could be dedicated to a specific TEE instance such as Intel SGX ECDSA,
+    AMD SEV-SNP, ARM Trustzone etc.
+    - It means that the client code wouldn’t need to support multiple
+      attestation protocols
+  - Backend TEE applications will still be able to run on various TEE machines,
+    since the attestation logic is implemented by the proxy
+
+## Attestation
+
+<!-- From: -->
+<!-- https://docs.google.com/drawings/d/1PUHf7NW3zV-zPz6zwTGUZI6wBVxy7yIfzxsbUbCw724 -->
+<img src="images/tls_attestation_ca.png" width="850">
+
+The key idea of Proxy Attestation Service is that it works as a **Certificate
+Authority (CA)** for backend applications. It means that after attesting backend
+applications it creates **signed TLS certificates** for them, which backend
```

Comment:
I think "backend applications" is not particularly understandable, unless somewhere first explains what the use case is (i.e. there are applications running in TEEs, and the CA not running in the TEE, and they obtain certs from the CA at startup time by performing remote attestation to the CA, etc.)